roofit/roofitcore/src/RooProduct.cxx

```diff
@@ -126,11 +126,11 @@ bool RooProduct::forceAnalyticalInt(const RooAbsArg& dep) const
   // Force internal handling of integration of given observable if any
   // of the product terms depend on it.
 
-  RooFIter compRIter = _compRSet.fwdIterator() ;
-  RooAbsReal* rcomp ;
   bool depends(false);
-  while((rcomp=(RooAbsReal*)compRIter.next())&&!depends) {
+  for (auto const* rcomp : static_range_cast<RooAbsReal*>(_compRSet)) {
+    if (!depends) {
         depends = rcomp->dependsOn(dep);
+    }
```

Comment:
```suggestion
  for (auto const* rcomp : static_range_cast<RooAbsReal*>(_compRSet)) {
    if (depends) break;
    depends = rcomp->dependsOn(dep);
```
The logic is not the same now. Previously, the look was stopped if `depends` is true, and now you just skip one iteration. You need to do something different, for example use `break`.